packages/codemirror-blocks/src/nodes.tsx

```diff
@@ -290,12 +357,13 @@ export function LambdaExpression(
     longDescription(node, level) {
       return `an anonymous function of ${pluralize(
         "argument",
-        node.fields.args.fields.ids
+        // TODO(pcardune): figure out how to describe node specs
+        // where a child must be a node of a particular type
+        node.fields.args.fields.ids as unknown[]
```

Comment:
So here's something interesting that popped up. Sometimes we have node types where the child nodes must be a particular type. nodeSpecs never did this deeper level of validation, and so its been possible to construct ASTs that validate but which blow up when they are rendered/described/prettied. Now the blowing up possibility is brought to light by typescript. I see two options:
1. Force functions that want to reach into a node's grand-children to do their own runtime validation or
2. provide a mechanism for enforcing this in the node spec.

For option 2, we then have to decide between nominal typing or structural typing. Do you say, "the args field must be a node of type identifierList" or do you say "the args field must be a node with an ids field which must be a list", or do you allow both. Then you also have the possibility of union types saying "the args field must be a node of one of the following types"

Reply:
I'm inclined to sat nominal typing. There are plenty of language constructs that might have identical fields but mean semantically different things. For example,  variable assignment and variable bindings, will both have `name` and `value` fields, but only one is allowed inside a function declaration.

Union types are intriguing. At that point, the nodeSpecs could almost be generated directly from a language's grammar.